src/plugins/usage_collection/public/services/create_reporter.ts

```diff
@@ -38,20 +38,20 @@ interface AnalyicsReporterConfig {
 }
 
 export function createReporter(config: AnalyicsReporterConfig): Reporter {
-  const { localStorage, debug } = config;
+  const { localStorage, debug, fetch } = config;
 
   return new Reporter({
     debug,
     storage: localStorage,
-    // async http(report) {
-    //   const response = await fetch.post('/api/ui_metric/report', {
```

Comment:
@ashwin-pc Do you know the reason why it was commented out previously? Is it related to a performance issue?

Reply:
I tried to hunt down he history here, but it looks like anything that seemed to send telemetry data was commented out during the fork out of caution and this was one of those. @kavilla can provide more context here but I doubt this had anything to do with performance since this does not add any significant overhead to the node server or its heap. If you can point me to any resources about that performance issue, we can try reproducing it to make sure that performance isnt impacted because of it. But otherwise i hope to merge this in sooner so that we can use the bake time to weed out any performance issues earlier on (if any)

Reply:
Collecting telemetry won't cause a load issue. Actually, when there are too many UI activities/requests to update the saved object, that can cause a significant load to OpenSearch

How about conduct some performance tests to collect baseline data so we could inform the customer about the performance impact (not OSD, but the target OpenSearch). We don't need too many tests; as minimal set, 1 concurrent user and 10 concurrent users, with each user potentially generating 1 UI event per second.

Understand that we might not have a performance test setup as part of CI. However, could we at least run some one-time tests to understand the baseline.

Another approach is to make it experimental (reminding users to enable it on the production environment at their own risk) if they want to enjoy this benefit.

As long as we don't introduce one-way door decisions, it is controllable behind a feature flag, and then it is up to the user to make the call.

Reply:
@seraphjiang @ashwin-pc Thanks for bringing up the point. UI metrics generated aren't saved immediately to the saved object, rather are batched in intervals and saved. If you are currently on a single page the metrics are batched in every 90s interval and then saved, however if you switch across pages the metrics gets saved at that the  timestamp when the switch takes place. Therefore it is highly unlikely we are generating a large number of requests per second.